### ISSUE TITLE
[Request]: Sweeps in mmdetection

### ISSUE THREAD

**Issue by David-Biggs** [wandb/examples#237]
Thanks for this amazing tool!! I have been blown away ever since I came across it.

I am using mmdetection to train my models. How do I go about setting my sweep within mmdetection?

Many thanks,
David

**Comment by ayulockin** [comment]
Hey @David-Biggs, thanks for the request. I will try to come up with something and let you know when I have an example.

Not sure if you are aware about this [PR](https://github.com/open-mmlab/mmdetection/pull/7459) that was recently merged in the `dev` branch of MMDetection that provides some dedicated support for MMDetection. 

For sweeps it might require some workaround but a proper integration would be better. I will try to scope it out.

**Comment by David-Biggs** [comment]
Hey @ayulockin 

Thanks a lot!

I will try this out ASAP ;)

**Comment by David-Biggs** [comment]
Hi @ayulockin,

So I've been working on it for a while and I made some alterations and have some interesting observations.

Alterations:

1. In train() you make use of `model = make_model(config)`. I removed this and used `model = build_detector(cfg.model)` (from mmdetection). I wasn't sure which one to use. I thought I would use the information in the 'current' sweep_config to update the cfg, by doing `cfg.optimizer.lr = config.learning_rate ... etc` , then pass `cfg.model` into `build_detector()`
2. I removed `meta=meta`. The default for meta is None and I wasn't sure where you defined your `meta` variable.

This works... Sort of. The model trains the sweeps loop over the different values but:

Obervations:
1. My training losses are all Nan
3. During training, I get this error `The testing results of the whole dataset is empty `. There are no validation results (mAP nor Losses)

I removed all sweep related code and ran the MMDetection `train_detector(model, datasets, cfg, distributed=False, validate=True)` command and it worked perfectly fine. Losses were real values and I got validation results. I did some digging but could not resolve either of the two issues.

**Comment by ayulockin** [comment]
Thanks for trying it out @David-Biggs. 

Sorry, I should have clarified that `make_model` was more of a pseudocode and not the actual API. Glad it worked (sort of).

Were you able to resolve the NaN loss issue?

**Comment by David-Biggs** [comment]
Hi @ayulockin,

So I found that the reason for the Nan loss issues was the learning rate. All the values I chose were slightly too large. After reducing the values I was able to do a successful sweep. 

Thanks again for your help.

**Comment by ayulockin** [comment]
Glad it worked for you. 👯‍♂️ 

Closing the issue since it's resolved.

**Comment by ayulockin** [comment]
@David-Biggs once I have my colab ready, I will share it here.
